lib/email_updater.rb

```diff
@@ -35,78 +27,103 @@ def change_to(email_input)
       end
     end
 
-    if errors.blank? && existing_user.nil?
-      args = {
-        old_email: @user.email,
-        new_email: email,
-      }
+    return if errors.present? || existing_user.present?
```

Comment:
Can we do the `return if errors.present?` earlier (eg. right after we validate the email) so that we avoid doing any work if the email is not valid?

Reply:
I can do that too.